### PR TITLE
fix: move single_shard_tracking.py to testloop and fix a flatstorage bug

### PR DESCRIPTION
**Pull request by Trisfald**
- Rewriting the test `single_shard_tracking` in testloop
- Fixing a bug uncovered by the new test

---

**Bug**: stale `StatePartsApplied` records cause memtrie assertion failure after crash during state sync

Prerequisites to trigger the bug:
- Node is killed mid-state-sync
- The DB has applied state parts, Ready flat storage and `DbCol::FlatState` values modified by post-sync block processing

On restart, catchup re-triggers state sync with the same sync_hash. The code sees `StatePartsApplied` records, assumes it's resuming a crashed mid-sync, and skips flat storage cleanup and part re-application. It then loads a memtrie from the dirty `DbCol::FlatState` using the original `state_root`, but the hashes don't match and it crashes.

Fix: if flat storage is already in memory (previous sync completed, state may be dirty), clear everything and re-apply from scratch.

### CODE REVIEW

File: test-loop-tests/src/tests/syncing.rs [hunk 1]
```diff
@@ -69,3 +74,110 @@ fn slow_test_sync_from_genesis() {
...
+        .epoch_config_store(epoch_config_store)
+        .clients(clients)
+        .gc_num_epochs_to_keep(20)
+        .build()
+        .warmup();
+
+    execute_money_transfers(&mut env.test_loop, &env.node_datas, &accounts).unwrap();
+    env.node_runner(0).run_for_number_of_blocks(20);
+
+    // Perform 4 kill-all/restart-all cycles.
```
Comment (pugachAG) [inline test-loop-tests/src/tests/syncing.rs]: why do we need exactly 4 cycles here? what kind of behavior is only triggered by that?

Comment (Trisfald) [reply test-loop-tests/src/tests/syncing.rs]: It roughly mirrors what the original test did to trigger multiple shard reassignment 
But to be honest 2 cycles would be enough to test: `restart -> shard shuffling ->  restart after restart -> shard shuffling`

Comment (pugachAG) [reply test-loop-tests/src/tests/syncing.rs]: let's reduce it to 2 cycles and add a comment mentioning why

Comment (Trisfald) [reply test-loop-tests/src/tests/syncing.rs]: Done
